cli/config/configfile/file.go

```diff
@@ -263,10 +264,26 @@ func decodeAuth(authStr string) (string, string, error) {
 // GetCredentialsStore returns a new credentials store from the settings in the
 // configuration file
 func (configFile *ConfigFile) GetCredentialsStore(registryHostname string) credentials.Store {
+	store := credentials.NewFileStore(configFile)
+
 	if helper := getConfiguredCredentialStore(configFile, registryHostname); helper != "" {
-		return newNativeStore(configFile, helper)
+		store = newNativeStore(configFile, helper)
+	}
+
+	// if DOCKER_AUTH_CONFIG is set, we need to use the env store instead
+	// it falls back to native or file store if a value is not found
+	// in the environment
+	if v := os.Getenv("DOCKER_AUTH_CONFIG"); v != "" {
+		envConfig := &ConfigFile{
+			AuthConfigs: make(map[string]types.AuthConfig),
+		}
```

Comment:
Curious; was it needed to initialise the `AuthConfigs` field? Looks like we could simplify;


```suggestion
		envConfig := &ConfigFile{}
```

I see I added a comment in the other implementation; not sure if useful here as well; https://github.com/docker/cli/blob/9e506545fdead79c574345a0414c71a0bd6857d6/cli/command/container/auth_config_utils.go#L27-L34